File: src/led/LEDDataBundle.java
package led;

import interpolation.InterpolationType;
import procedures.ProcedureCalls;
import procedures.models.Direction;

public class LEDDataBundle {
	public ColorRGB colorPrimary;
	public ColorRGB colorSecondary;
	public ColorRGB colorTertiary;
	
	public String data;
	
	public Float value1;
	public Float value2;
	public Float value3;
	
	public Integer modulo;
	public InterpolationType interpolation;
	public Direction direction;
	
	public Integer bpm;
	public Float repetitions;
	public Float speed;
	public Integer duration;
	public Boolean pulsating;
	public String path;
	//subBundle
	public Boolean isSubProcedure;
	
	public Boolean indeterminate;
	
	public Integer puModulo;
	public Boolean puModuloInvert;
	
	public LEDStripManager ledStrip;
	public ProcedureCalls procedureCalls;
}

File: src/main/java/led/LEDDataBundle.kt
package led

import com.github.iamniklas.liocore.interpolation.*
import procedures.ProcedureCalls
import procedures.models.Direction

class LEDDataBundle {
    @JvmField
    var colorPrimary: ColorRGB? = null
    @JvmField
    var colorSecondary: ColorRGB? = null
    var colorTertiary: ColorRGB? = null
    @JvmField
    var data: String? = null
    @JvmField
    var value1: Float? = null
    @JvmField
    var value2: Float? = null
    var value3: Float? = null
    @JvmField
    var modulo: Int? = null
    var interpolation: InterpolationType? = null
    @JvmField
    var direction: Direction? = null
    var bpm: Int? = null
    @JvmField
    var repetitions: Float? = null
    @JvmField
    var speed: Float? = null
    @JvmField
    var duration: Int? = null
    @JvmField
    var pulsating: Boolean? = null
    @JvmField
    var path: String? = null

    //subBundle
    var isSubProcedure: Boolean? = null
    @JvmField
    var indeterminate: Boolean? = null
    @JvmField
    var puModulo: Int? = null
    @JvmField
    var puModuloInvert: Boolean? = null
    @JvmField
    var ledStrip: LEDStripManager? = null
    @JvmField
    var procedureCalls: ProcedureCalls? = null
}